File: MasterCatalog.Web/wwwroot/js/models/item.js
export class Item {
    itemID;
    upc;
    upcError;
    itemName;
    itemNameError;
    itemDescription;
    unitOfMeasureID;
    unitOfMeasureIDError;
    unitQuantity;
    unitQuantityError;
    categoryID;
    categoryIDError;
    attributes;
    images;
    unitOfMeasure;
    category;


    constructor(id, name, description, qty) {
        this.itemID = id;
        this.itemName = name;
        this.itemDescription = description;
        this.unitQuantity = qty;
    }

    validate() {
        this.clearErrors();

        let valid = true;
        const name = this.itemName;
        if (!name) {
            this.itemNameError = 'Item name is Required.';
            valid = false;
        }

        const upc = this.upc;
        if (!upc) {
            this.upcError = 'Upc is Required.';
            valid = false;
        }

        const qty = this.unitQuantity;
        if (!qty) {
            this.unitQuantityError = 'Quantity is Required.';
            valid = false;
        }

        const uom = this.unitOfMeasureID;
        if (!uom) {
            this.unitOfMeasureIDError = 'Unit of Measure is Required.';
            valid = false;
        }

        const category = this.categoryID;
        if (!category) {
            this.categoryIDError = 'Category is Required.';
            valid = false;
        }

        return valid;
    }

    clearErrors() {
        this.itemNameError = null;
        this.upcError = null;
        this.unitQuantityError = null;
        this.unitOfMeasureIDError = null;
        this.categoryIDError = null;
    }
}

File: MasterCatalog.Domain/Models/Item.cs
using System.ComponentModel.DataAnnotations;

namespace MasterCatalog.Domain.Models
{
    public class Item
    {
        public Item()
        {
            Attributes = new List<ItemAttribute>();
            Images = new List<ItemImage>();
        }

        public int? ItemID { get; set; }

        [Required(AllowEmptyStrings = false, ErrorMessage = "Upc is required")]
        public string Upc { set; get; }


        [Required(AllowEmptyStrings = false, ErrorMessage = "Item name is required")]
        public string ItemName { get; set; }

        public string ItemDescription { get; set; }

        [DeniedValues(0, ErrorMessage = "UOM identifier is required")]
        public int UnitOfMeasureID { set; get; }

        [DeniedValues(0, ErrorMessage = "Item quantity is required")]
        public double UnitQuantity { set; get; }

        public int CategoryID { set; get; }

        public List<ItemAttribute> Attributes { get; set; }

        public Category? Category { get; set; }

        public UnitOfMeasure? UnitOfMeasure { get; set; }

        public List<ItemImage> Images { get; set; }
    }
}
